crypto/initthread.c

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/crypto.h>
+#include <openssl/core_numbers.h>
+#include "internal/cryptlib_int.h"
+#include "internal/providercommon.h"
+
+#ifdef FIPS_MODE
+extern OSSL_core_thread_start_fn *c_thread_start;
+#endif
```

Comment:
This is a hack that needs some serious commenting.  It took me quite a bit of looking back and forth before I could clear out most "Huh?" and "WTa*?" from my mind...  I quite frankly dislike this, but I can't currently see another way.

Reply:
I added a big comment here explaining what is going on.